### PR TITLE
chore: Update commit validation to use commitlint

### LINKED ISSUE #62: Update to commitlint
## 🚀 Feature Proposal

Move from `validate-commit-msg` to `commitlint`

## Motivation

`validate-commit-msg` has been deprecated in favor of `commitlint`: https://www.npmjs.com/package/validate-commit-msg. It's working fine as is though, so this is a low priority

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #62

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -130,5 +130,8 @@
   "workspaces": [
     "modules/**"
   ],
-  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11"
+  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11",
+  "dependencies": {
```
Comment (lychyi) [inline package.json]: This changeset is odd, did you add `commitlint` to dependencies and then remove?

**Comment by Patil2099** [comment]
I added commitlint to devdependencies and removed it from there

On Aug 20, 2019 3:29 AM, Lynn Chyi <notifications@github.com> wrote:

@lychyi commented on this pull request.

________________________________

In package.json<https://github.com/Workday/canvas-kit/pull/124#discussion_r315430457>:

> @@ -130,5 +130,8 @@
   "workspaces": [
     "modules/**"
   ],
-  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11"
+  "browserslist": "> 5%, last 2 firefox versions, last 2 chrome versions, last 2 safari versions, last 2 edge versions, ie 11",
+  "dependencies": {


This changeset is odd, did you add commitlint to dependencies and then remove?

File: package.json (hunk 1 above)
Comment (NicholasBoll) [reply package.json]: Agreed. I can fix this and merge today

**Comment by NicholasBoll** [comment]
This PR isn't complete. When creating a commit, the command `npm run -s commitmsg` is run first which right now runs `validate-commit-msg` which no longer exists. The `commitmsg` command in `package.json` needs to be updated to handle linting correctly.

**Comment by NicholasBoll** [comment]
I'm following https://github.com/conventional-changelog/commitlint#getting-started to see how `commitlint` works

**Comment by NicholasBoll** [comment]
I finished this PR. I'll let the team decide on the commit message format. `commitlint` is more strict than what we had before. It follows https://www.conventionalcommits.org/en/v1.0.0-beta.4/:

Before:
```
# No scope checks, can have sentence-case
fix(WhatEverYouWant): Fix something here
```

After
```
# Scope is checked against `modules` folder, cannot be sentence-case
fix(button): remove extra padding
```

Not a breaking change.

### Suggested squash message:
chore: update commit validation to use commitlint

**Note**
Squash commit messages are not checked!

**Pull request merged**